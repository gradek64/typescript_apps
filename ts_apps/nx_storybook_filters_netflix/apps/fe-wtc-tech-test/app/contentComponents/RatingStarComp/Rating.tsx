import React from 'react';
import styled from 'styled-components';
//custom
import { showStarRating } from './RatingStarUtills/ratingCalculation';

const RatingContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: 0 10px;
`;

interface PropsInt {
  rating: number;
}

function Rating(props: PropsInt) {
  const { rating } = props;
  const allStars = 5;
  const stars = showStarRating(rating, allStars);
  const percentage = (rating * allStars) / 100;
  return (
    <RatingContainer>
      {stars.map((Comp, i) => (
        <Comp key={`star${i}`} />
      ))}
      <span>({String(percentage.toFixed(2))})</span>
    </RatingContainer>
  );
}

export default Rating;
