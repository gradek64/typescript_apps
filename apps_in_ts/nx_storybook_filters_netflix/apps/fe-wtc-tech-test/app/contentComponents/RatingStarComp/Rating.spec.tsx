import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Rating from './Rating';

const props = {
  rating: 50,
};

describe('should render correclty with initial state', () => {
  afterEach(cleanup);
  it('should render and take a snapshot', () => {
    const { asFragment } = render(<Rating {...props} />);
    const component = asFragment();
    expect(component).toMatchSnapshot();
  });
});
