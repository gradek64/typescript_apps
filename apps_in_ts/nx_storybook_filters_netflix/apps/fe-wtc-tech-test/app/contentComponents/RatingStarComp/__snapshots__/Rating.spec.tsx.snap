// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correclty with initial state should render and take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm laBNJz"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4V6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
      />
    </svg>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
      />
    </svg>
    <span>
      (2.50)
    </span>
  </div>
</DocumentFragment>
`;
